File: 1021-Remove-Outermost-Parentheses.py
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        out = []
        count = 0
        for s in S:
            if s == ')':
                count -= 1
            if count > 0:
                out.append(s)
            if s == '(':
                count += 1
        return "".join(out)
        
        
class SolutionII:
    def removeOuterParentheses(self, S: str) -> str:
        out = []
        count = 0
        for s in S:
            if s == '(' and count > 0: out.append(s)
            elif s == ')' and count > 1: out.append(s)
            count += 1 if s == '(' else -1
        return "".join(out)

File: 1021-Remove-Outermost-Parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string S) {
        string out;
        int count = 0;
        for (char s : S){
            if(s == '(' && count++ > 0 ) out += s;
            if(s == ')' && count-- > 1 ) out += s;
        }
        return out;
    }
};
